src/tutorials/pet-shop.md

```diff
@@ -374,6 +376,100 @@ Since arrays can only return a single value given a single key, we create our ow
+      before("adopt a pet using accounts[0]", async () => {
+      await adoption.adopt(8, { from: accounts[0] });
+      expectedAdopter = accounts[0];
+    });
+    });
+  });
+
+  ```
+  We start the contract by importing : 
+  * ```Adoption.sol```:The smart contract we want to test
+  * ```chai```: Gives us various assertions to use in our tests to be used for equality, inequality or emptiness to return a pass/fail from our test.
+
+  **Note**: When writing the contract, we make use of the callback function ```accounts``` that provides us with the accounts in the development or the test network used.
+
+  Then, we make use of async functions : 
+  * To deploy the contract as soon as the test begins for the contract 
+  * To use the asynchronous feature to get the address of the contract and expectedAdopter
+
+  ### Testing the adopt() function 
+
+  To test the ```adopt()``` function, recall that upon success it returns the given ```petId```. We can ensure an ID was returned and that it's correct by comparing the return value of ```adopt()``` to the ID we passed in.
+
+  1. Add the following function within the ```testAdoption.test.js``` test file, after the declaration of ```describe```. 
+
+  ```
+  describe("adopting a pet and retrieving account addresses", async () => {
+    before("adopt a pet using accounts[0]", async () => {
+      await adoption.adopt(8, { from: accounts[0] });
+      expectedAdopter = accounts[0];
+    });
+
+    it("can fetch the address of an owner by pet id", async () => {
+      const adopter = await adoption.adopters(8);
+      assert.equal(adopter, expectedAdopter, "The owner of the adopted pet should be the first account.");
+    });
+  });
+  ```
+
+  Things to notice:
+
+  * We call the smart contract we declared earlier with the ID of expectedPetId.
+  * When called the function within the smart contract, we send metadata for the solidity contract to know that ```msg.sender``` is ```accounts[0]```.
+  * Finally, we pass the actual value, the expected value and a failure message (which gets printed to the console if the test does not pass) to assert.equal().
+
+  ### Testing retrieval of a single pet's owner
+
+  1. Add this function below the previously added function in ```testAdoption.test.js```.
+
+  ```
+  it("can fetch the address of an owner by pet id", async () => {
+    const adopter = await adoption.adopters(8);
+    assert.equal(adopter, expectedAdopter, "The owner of the adopted pet should be the first account.");
+  });
+  ```
+
+  After getting the adopter address stored by the adoption contract, as assert equality as we did above.
+
+  ### Testing retrieval of all pet owners
+
+  Since arrays can only return a single value given a single key, we create our own getter for the entire array.
+
+  1. Add this function below the previously added function in ```testAdoption.test.js```.
+  
+  ```
+  it("can fetch the collection of all pet owners addresses", async () => {
```

Comment:
Sorry I made a grammar mistake. This should be "...of all pet owners' addresses"